### PR TITLE
Update external contributing guidelines

### PR DESCRIPTION
**Pull request by andrewsavage1**
Note that this mainly follows tensorflow's contributing guidelines, found [here](https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kaidokert** [comment]
Looks like a good pass. Just one question - was there anything on OSPO guidelines for the content here ? I think they have one referenced ( look for `http://cs/f:teams/opensource/releasing/template` ). More language is better, but we should ensure  we aren't leaving anything out.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,64 +1,101 @@
...
+    all compiler warning in a file, etc.)
+
+#### License
+
+Include a license at the top of new files. Check existing files for license examples.
+
+#### Coding style
+
+Changes to Cobalt C++ code should conform to
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
```
Comment (kaidokert) [inline CONTRIBUTING.md:68]: Shouldn't this be Chrome/Chromium page ? https://chromium.googlesource.com/chromium/src/+/main/styleguide/styleguide.md

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,64 +1,101 @@
...
+Changes to Cobalt Python code should conform to
+[Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md)
+
+
+#### Coding style
+
+Cobalt uses pre-commit to ensure good coding style. Create a python 3 virtual
```
Comment (kaidokert) [inline CONTRIBUTING.md:75]: Let's put a link under [pre-commit](`https://pre-commit.com/)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,64 +1,101 @@
...
+You can find documentation about it at https://pre-commit.com/.
+
+*   [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html)
+*   [Google Python Style Guide](https://google.github.io/styleguide/blob/gh-pages/pyguide.html)
+*   [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)
+*   [Google JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html)
+*   [Google Shell Style Guide](https://google.github.io/styleguide/shellguide.html)
+*   [Google Objective-C Style Guide](https://google.github.io/styleguide/objcguide.html)
+
+#### Running unit tests
```
Comment (kaidokert) [inline CONTRIBUTING.md:92]: Should we say anything about what to expect from CI runs ? E.g. that external contributor CI runs need to be approved, that style checks are enforced, and all checks are expected to be green before merging.

Also we don't, for now, set any expectations around review response rates or that the code will be accepted.

File: CONTRIBUTING.md (hunk 2 above)
Comment (andrewsavage1) [reply CONTRIBUTING.md:75]: The link is in the next sentence

File: CONTRIBUTING.md (hunk 1 above)
Comment (andrewsavage1) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,64 +1,94 @@
...
+## Pull Request Checklist
 
+Before sending your pull requests, make sure you do the following:
 
-## Before You Contribute
+-   Read the [contributing guidelines](CONTRIBUTING.md).
+-   Ensure you have signed the
+    [Contributor License Agreement (CLA)](https://cla.developers.google.com/).
+-   Check if your changes are consistent with the
+    [guidelines](#general-guidelines-and-philosophy-for-contribution).
```
Comment (alexanderahn) [inline CONTRIBUTING.md:11]: Suggest putting [guidelines](#general-guidelines-and-philosophy-for-contribution) and [Coding Style](#coding-style) in a lower level unordered list. So

-  Check if your changes are consistent with the:
    -  [General guidelines](#general-guidelines-and-philosophy-for-contribution)
    -  [Coding Style](#coding-style)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,64 +1,94 @@
...
-     the change into our private repository, and it will get pushed out to the
-     public repository after some time.
+Before sending your pull request for
+[review](https://github.com/tensorflow/tensorflow/pulls),
+make sure your changes are consistent with the guidelines and follow the
+Cobalt coding style.
+
+#### General guidelines and philosophy for contribution
+
+*   Include unit tests when you contribute new features, as they help to a)
```
Comment (alexanderahn) [inline CONTRIBUTING.md:52]: I would do the same here with another level of unordered list if possible. Just make it easier to read.

File: CONTRIBUTING.md (hunk 3 above)
Comment (andrewsavage1) [reply CONTRIBUTING.md:85]: I do mention that checks should pass in "Contributing code," but I made it a bit more pointed

**Review by alexanderahn — APPROVED** [review]

**Pull request merged**